src/main/java/core/basesyntax/supplier/ShapeSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.shape.Circle;
+import core.basesyntax.shape.IsoscelesTrapezoid;
+import core.basesyntax.shape.Rectangle;
+import core.basesyntax.shape.RightTriangle;
+import core.basesyntax.shape.Shape;
+import core.basesyntax.shape.Square;
+import core.basesyntax.type.Color;
+
+public class ShapeSupplier extends BasicRandomValueSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    @Override
+    public Shape getRandom() {
+        Color color = colorSupplier.getRandom();
+        double randomDouble1 = randomGenerator.nextDouble();
```

Comment:
Consider giving more descriptive names to randomDouble1, randomDouble2, and randomDouble3, such as sideLength1, sideLength2, and sideLength3.